btree/btree.c

```diff
@@ -3733,6 +3755,9 @@ struct cb_data {
 
 static int btree_kv_put_cb(struct m0_btree_cb *cb, struct m0_btree_rec *rec)
 {
+	if (rec->r_flags == M0_BSC_KEY_EXISTS) {
+		return 0;
```

Comment:
You need to return a non-zero value in case of error so that the caller can handle it.